static_src/actions/space_actions.js

```diff
@@ -5,27 +5,41 @@
  */
 
 import AppDispatcher from '../dispatcher.js';
+import cfApi from '../util/cf_api';
 import { spaceActionTypes } from '../constants.js';
+import SpaceStore from '../stores/space_store';
 
 export default {
   fetch(spaceGuid) {
     AppDispatcher.handleViewAction({
       type: spaceActionTypes.SPACE_FETCH,
       spaceGuid
     });
+
+    // TODO error action should also be specified here
+    return cfApi.fetchSpace(spaceGuid).then(this.receivedSpace);
```

Comment:
I see. I think that by putting the `receivedSpace` action in the cfApi, we open ourselves up to some action mismatches. It becomes possible to want to fetch from the space API, which would trigger the `RECEIVE`, but it's on the api caller to call the `FETCH` leading to possible `RECEIVE` without the initial `FETCH`.

By moving it all to the action, the `FETCH`, `RECEIVE`, and `ERROR` would all be located in the same action creation and decoupled from the actual cfApi call. e.g.
```javascript
function fetchAction() {
  dispatch(FETCH)
  cfApi.fetch().then(() => dispatch(RECEIVE)).catch(() => dispatch(ERROR)).
}
```
I think this makes it much clearer who is responsible for what:
- cfApi: handles API requests and data parsing/formatting
- actions: handle the async flow of cfApi and encapsulate the dispatch of action events
- stores: synchronously updates the loading state based on the action events.